frontend/src/metabase/search/containers/SearchApp.unit.spec.tsx

```diff
@@ -0,0 +1,212 @@
+      initialRoute,
+    },
+  );
+
+  await waitFor(() => {
+    expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument();
+  });
+
+  return {
+    history,
+  };
+};
+
+describe("SearchApp", () => {
+  describe("rendering search results and pagination", () => {
+    it("renders empty state when there are no search results", async () => {
+      // let's pick some text that doesn't ever match anything
+      await setup({ searchText: "oisin" });
+
+      expect(screen.getByText('Results for "oisin"')).toBeInTheDocument();
+      expect(
+        screen.getByText("There weren't any results for your search."),
+      ).toBeInTheDocument();
+      expect(screen.queryByLabelText("pagination")).not.toBeInTheDocument();
+    });
+
+    it("renders search results when there is at least one result", async () => {
+      await setup({ searchText: "Card" });
+
+      const searchResultsHeader = screen.getByText('Results for "Card"');
+      expect(searchResultsHeader).toBeInTheDocument();
+
+      const searchResults = screen.getAllByTestId("search-result-item");
+      expect(searchResults.length).toEqual(1);
+      expect(screen.queryByLabelText("pagination")).not.toBeInTheDocument();
+    });
+
+    it("renders search results and pagination when there is more than PAGE_SIZE results", async () => {
+      await setup({ searchText: "a" });
+      const getPaginationTotal = () => screen.getByTestId("pagination-total");
+      const getPagination = () => screen.getByLabelText("pagination");
+      const getNextPageButton = () => screen.getByTestId("next-page-btn");
+      const getPreviousPageButton = () =>
+        screen.getByTestId("previous-page-btn");
+
+      expect(getPaginationTotal()).toHaveTextContent("5");
+      expect(getPreviousPageButton()).toBeDisabled();
+      expect(getNextPageButton()).toBeEnabled();
+      expect(getPagination()).toHaveTextContent("1 - 3");
+
+      // test next page button
+      userEvent.click(getNextPageButton());
+      await waitFor(() => {
+        expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument();
+      });
+      expect(getPaginationTotal()).toHaveTextContent("5");
+      expect(getPreviousPageButton()).toBeEnabled();
+      expect(getNextPageButton()).toBeDisabled();
+
+      expect(getPagination()).toHaveTextContent("4 - 5");
+
+      // test previous page button
+      userEvent.click(getPreviousPageButton());
+      await waitFor(() => {
+        expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument();
+      });
+      expect(getPaginationTotal()).toHaveTextContent("5");
+      expect(getPreviousPageButton()).toBeDisabled();
+      expect(getNextPageButton()).toBeEnabled();
+      expect(getPagination()).toHaveTextContent("1 - 3");
+    });
+  });
+
+  describe("filtering search results with the sidebar", () => {
+    it("should reload with filtered searches when a type on the right sidebar is clicked without changing URL", async () => {
```

Comment:
Can we add tests that click all the options in the sidebar too? This only test clicking `Questions`

Reply:
Yep, I've modified the test to load all entities, then click through all items in the sidebar, checking that search results only contain that entity